### PR TITLE
UNet for ISIC 2018 Image Segmentation

### PR DESCRIPTION
**Pull request by s4542006**
This solution set has four modules each containing several functions for loading image data, defining, training and evaluating UNet for image segmentation. To train the UNet model and output its predictive performance metric, only driver_script.py module, which contains the main() function, should be run that will automatically call all necessary functions. This will ask for the following user inputs: (i) if the model to be trained in 50 (default value) epochs or the user may opt to change the number. Then, input and ground truth image directories (full path) have to be provided. All necessary details about the model and dependencies are included in the README.md file. Besides, the learning curve and sample images are also added.

Name: Md Abdul Bari
ID: 45420065

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Sufficiently detailed commit messages
- Code valid for Easy Difficulty
- Results are good

Other Feedback
- Please group the `.png` files and put them in a sub-directory
- final batch norm before activation not necessary

**Comment by rexsw** [comment]
Code looks good for easy difficulty

For next time;

There’s no need to use a whole library for one function rather than using pandas you could have written that function yourself

Tensorflow has functions for image resizing and more so there’s no need to also use skimage in your project. Same for sklearn and train/test splitting

**Comment by s4542006** [comment]
As per tutor's feedback, following changes were made:
1. all image (.png) files were moved to a separate sub-directory
2. batch norm before the final activation was dropped

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. Move image files into sub directory.**
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Comment by s4542006** [comment]
Hi Shakes,

Based on tutor's  feedback, I already moved image files  into a separate sub-directory that's also mentioned in your 'feedback required'. Thus, I believed I've already addressed your feedback and my branch is now ready to be merged with yours. Please let me know if I need to do anything else in this connection. Thanks in advance.
